core/src/commonMain/kotlin/com/ustadmobile/core/controller/ContentEntryDetailAttemptsListPresenter.kt

```diff
@@ -0,0 +1,84 @@
+package com.ustadmobile.core.controller
+
+import com.ustadmobile.core.db.dao.StatementDao
+import com.ustadmobile.core.generated.locale.MessageID
+import com.ustadmobile.core.util.SortOrderOption
+import com.ustadmobile.core.util.ext.toQueryLikeParam
+import com.ustadmobile.core.view.ContentEntryDetailAttemptsListView
+import com.ustadmobile.core.view.SessionsListView
+import com.ustadmobile.core.view.UstadView
+import com.ustadmobile.door.DoorLifecycleOwner
+import com.ustadmobile.lib.db.entities.PersonWithStatementDisplay
+import com.ustadmobile.lib.db.entities.UmAccount
+import org.kodein.di.DI
+
+class ContentEntryDetailAttemptsListPresenter(context: Any, arguments: Map<String, String>,
+                                              view: ContentEntryDetailAttemptsListView,
+                                              di: DI, lifecycleOwner: DoorLifecycleOwner)
+    : UstadListPresenter<ContentEntryDetailAttemptsListView, PersonWithStatementDisplay>(
+        context, arguments, view, di, lifecycleOwner), OnSortOptionSelected, OnSearchSubmitted {
+
+    private var contentEntryUid: Long = 0L
+
+    var searchText: String? = null
+
+    override val sortOptions: List<SortOrderOption>
+        get() = SORT_OPTIONS
+
+    override fun onCreate(savedState: Map<String, String>?) {
+        super.onCreate(savedState)
+        selectedSortOption = SORT_OPTIONS[0]
+        contentEntryUid = arguments[UstadView.ARG_ENTITY_UID]?.toLong() ?: 0
+    }
+
+    override suspend fun onCheckAddPermission(account: UmAccount?): Boolean {
+        return false
+    }
+
+    override suspend fun onLoadFromDb() {
+        super.onLoadFromDb()
+        mLoggedInPersonUid = accountManager.activeAccount.personUid
+        updateListOnView()
+    }
+
+    override fun onClickSort(sortOption: SortOrderOption) {
+        super.onClickSort(sortOption)
+        updateListOnView()
+    }
+
+    private fun updateListOnView() {
+        view.list = repo.statementDao.findPersonsWithContentEntryAttempts(contentEntryUid,
+                mLoggedInPersonUid, searchText.toQueryLikeParam(),
+                selectedSortOption?.flag ?: StatementDao.SORT_FIRST_NAME_ASC)
+    }
+
+    override fun onSearchSubmitted(text: String?) {
+        searchText = text
+        updateListOnView()
+    }
+
+
+    override fun handleClickCreateNewFab() {
+
+    }
+
+    fun onClickPersonWithStatementDisplay(personWithStatementDisplay: PersonWithStatementDisplay) {
+            systemImpl.go(SessionsListView.VIEW_NAME,
```

Comment:
Why are we passing through all the arguments? That could lead to unpredictable behavior. We should only be passing arguments through for specific use cases (e.g. authentication login flow etc).  SessionListView should take ARG_CONTENT_ENTRY_UID and ARG_PERSON_UID.